### PR TITLE
Refactor acceptance test

### LINKED ISSUE #324: Acceptance test not reflect correct behavior
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 7,8
- Ruby: *
- Distribution: Debian 12 EL9 Ubuntu 22.04
- Module version: 4.1.0

## How to reproduce (e.g Puppet code you use)
start keepalived service without configuration on new versione with updated unit-file

## What are you seeing
Service fail to start in acceptance test when try to run service without minimal configuration of vrrp_instance

## Output log
```
  Mar 12 18:26:19 debian12-64-puppet8.example.com systemd[1]: keepalived.service - Keepalive Daemon (LVS and VRRP) was skipped because of an unmet condition check (ConditionFileNotEmpty=/etc/keepalived/keepalived.conf).
  Mar 12 18:26:20 debian12-64-puppet8.example.com systemd[1]: Starting keepalived.service - Keepalive Daemon (LVS and VRRP)...
  Mar 12 18:26:20 debian12-64-puppet8.example.com Keepalived[1347]: Starting Keepalived v2.2.7 (01/16,2022)
  Mar 12 18:26:20 debian12-64-puppet8.example.com Keepalived[1347]: WARNING - keepalived was built for newer Linux 5.19.11, running on Linux 5.15.0-1057-azure #65~20.04.1-Ubuntu SMP Mon Feb 12 17:26:40 UTC 2024
  Mar 12 18:26:20 debian12-64-puppet8.example.com Keepalived[1347]: Command line: '/usr/sbin/keepalived' '--dont-fork' '-D' '--vrrp'
  Mar 12 18:26:20 debian12-64-puppet8.example.com Keepalived[1347]: Opening file '/etc/keepalived/keepalived.conf'.
  Mar 12 18:26:20 debian12-64-puppet8.example.com Keepalived[1347]: Configuration file /etc/keepalived/keepalived.conf
  Mar 12 18:26:20 debian12-64-puppet8.example.com Keepalived[1347]: NOTICE: setting config option max_auto_priority should result in better keepalived performance
  Mar 12 18:26:20 debian12-64-puppet8.example.com Keepalived[1347]: Warning - keepalived has no configuration to run
  Mar 12 18:27:50 debian12-64-puppet8.example.com systemd[1]: keepalived.service: start operation timed out. Terminating.
  Mar 12 18:27:50 debian12-64-puppet8.example.com Keepalived[1347]: Stopping
  Mar 12 18:27:50 debian12-64-puppet8.example.com Keepalived[1347]: CPU usage (self/children) user: 0.005619/0.000000 system: 0.000000/0.000000
  Mar 12 18:27:50 debian12-64-puppet8.example.com Keepalived[1347]: Stopped Keepalived v2.2.7 (01/16,2022)
  Mar 12 18:27:50 debian12-64-puppet8.example.com systemd[1]: keepalived.service: Failed with result 'timeout'.
  Mar 12 18:27:50 debian12-64-puppet8.example.com systemd[1]: Failed to start keepalived.service - Keepalive Daemon (LVS and VRRP).
```

## Any additional information you'd like to impart
Unit-file non new distribution change Service Type from `simple` to `notify`
As expected keepalived not start without a valid configuration.

#296 #299 #297

### PR DESCRIPTION
**Pull request by Valantin**
#### Pull Request (PR) description
Use shared example for idempotency test
setup host with minimal working configuration

#### This Pull Request (PR) fixes the following issues
Fixes #324

### CODE REVIEW

File: spec/acceptance/keepalived_spec.rb [hunk 1]
```diff
@@ -78,63 +40,94 @@ class { 'keepalived':
...
+  context 'with unmanaged external config' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-PUPPET
+        file { '/etc/keepalived/myconfig.conf':
+          owner   => 'root',
+          group   => 'root',
+          mode    => '0644',
+          content => "vrrp_instance VI_50 { interface ${facts['networking']['primary']}
+          virtual_router_id 50 }",
```
Comment (bastelfreak) [inline spec/acceptance/keepalived_spec.rb:90]: the newline in the string looks a bit odd. Is that on purpose?

Comment (Valantin) [reply spec/acceptance/keepalived_spec.rb:90]: I don't know what separator keepalive use for different configuration, I've tried comma and semicolon but not work, with newline it pass the check during service start

Comment (jay7x) [reply spec/acceptance/keepalived_spec.rb:90]: "...\n..." should be able to do the trick w/o wrapping the line then (I *guess*)

File: spec/acceptance/keepalived_spec.rb [hunk 2]
```diff
@@ -78,63 +40,94 @@ class { 'keepalived':
     end
 
     # Works around any timing issues
-    it 'has acquired the ip' do
-      ip_result = shell('sleep 10; ip addr')
-      expect(ip_result.stdout).to match %r{.*inet 10\.0\.0\.1/16 .*}
+    describe 'when service running' do
+      it 'has acquired the ip' do
+        ip_result = shell('sleep 10; ip addr')
```
[the remark this answers is not in the document]
Comment (Valantin) [reply spec/acceptance/keepalived_spec.rb:45]: thanks for suggestion but I'm try now and in this way it fails, don't know why

**Pull request merged**